rasa/nlu/training_data/formats/rasa_yaml.py

```diff
@@ -0,0 +1,189 @@
+import logging
+from typing import Text, Any, List, Dict, Tuple
+
+import typing
+
+from rasa.nlu.training_data.entities_parser import EntitiesParser
+from rasa.nlu.training_data.formats.readerwriter import TrainingDataReader
+import rasa.utils.io
+from rasa.nlu.training_data.lookup_tables_parser import LookupTablesParser
+from rasa.nlu.training_data.synonyms_parser import SynonymsParser
+
+if typing.TYPE_CHECKING:
+    from rasa.nlu.training_data import TrainingData
+
+
+logger = logging.getLogger(__name__)
+
+KEY_NLU = "nlu"
+KEY_INTENT = "intent"
+KEY_INTENT_EXAMPLES = "examples"
+KEY_INTENT_TEXT = "text"
+KEY_SYNONYM = "synonym"
+KEY_SYNONYM_EXAMPLES = "examples"
+KEY_REGEX = "regex"
+KEY_REGEX_EXAMPLES = "examples"
+KEY_LOOKUP = "lookup"
+KEY_LOOKUP_EXAMPLES = "examples"
+
+DOCS_LINK = "https://rasa.com/docs/rasa/"
+
+
+class RasaYAMLReader(TrainingDataReader):
+    def __init__(self) -> None:
+        self.training_examples = []
+        self.entity_synonyms = {}
+        self.regex_features = []
+        self.lookup_tables = []
+
+    def reads(self, s: Text, **kwargs: Any) -> "TrainingData":
+        from rasa.nlu.training_data import TrainingData
+
+        self.__init__()
```

Comment:
Calling `__init__` explicitly might be a bit confusing to read. We could move the code from `__init__` into a method called `_initialize`, and call it from there and here.

Reply:
👍